backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/StubRenderer.kt

```diff
@@ -6,18 +6,32 @@
 package org.jetbrains.kotlin.backend.konan.objcexport
 
 import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.descriptors.DeclarationDescriptor
+import org.jetbrains.kotlin.descriptors.DeclarationDescriptorWithSource
+import org.jetbrains.kotlin.psi.KtDeclaration
+import org.jetbrains.kotlin.psi.KtPrimaryConstructor
+import org.jetbrains.kotlin.resolve.source.getPsi
+import org.jetbrains.kotlin.backend.common.serialization.extractSerializedKdocString
 
 object StubRenderer {
     fun render(stub: Stub<*>): List<String> = collect {
         stub.run {
+            val kDocString = this.descriptor?.extractKDocString()
             this.comment?.let { comment ->
                 +"" // Probably makes the output more readable.
                 +"/**"
+                kDocString?.let {
+                    + it.removeSurrounding("/**", "*/")
+                    +"" // Probably makes the following comment more readable.
+                }
                 comment.contentLines.forEach {
                     +" $it"
                 }
                 +"*/"
             }
+                    // Let's try to keep non-trivial kdoc formatting intact, when there is no other comments here
+                    ?: kDocString?.let { +it }
```

Comment:
Generally the resulting list is a list of lines, while this code might add a "multiline" string to it. This is probably not critical, but following the established convention is still desirable.